### DISCUSSION TITLE
Recruitment

### DISCUSSION THREAD

**Discussion by JunyanDing** [NGEET/fates#1323] [Ideas]
The current recruitment model calculates the number of individuals recruited by dividing the total mass of germinated seeds by the mass of the individual plant to be recruited. 
https://github.com/NGEET/fates/blob/master/biogeochem/EDPhysiologyMod.F90#L1804
https://github.com/NGEET/fates/blob/master/biogeochem/EDPhysiologyMod.F90#L1970

Conceptually, this means that a recruited plant is made by a bunch of seeds. I feel this is not appropriate. Because it could largely underestimate the recruitment rate. It also makes the rare PFT easier to go extinct. Further, this makes the recruitment rate rely on the initial size of the plant, which can have an effect on multi PFTs simulation, e.g. size asymmetric PFTs. Not sure to which extent this can affect the tree-grass interaction and consequently the fire. I could be totally wrong. @rgknox , please correct me if this is not the case.

Suggested changes: introduce a new parameter, average seed mass of a single seed, then divide the total germinated seeds mass by the mass of single seed to generate the number of recruiting plants. Use soil pool for nutrient and water needed for the recruitment, whereas carbon should come from atmosphere.

**Comment by mpaiao** [comment]
@JunyanDing I think you could effectively simulate this effect by reducing the minimum height, without adding a new parameter.

As you correctly pointed out, the way reproduction is represented does not reflect how it happens in nature. Just to put your question in a "historical" context, the reproduction routine seems to be based on ED1.  The reason for this was mostly practical: seedling biomass may be extremely low, which may cause numerical instabilities (maybe less of an issue in FATES), and allometric equations are very likely off by orders of magnitude for very small individuals, which may make seedling viability rather challenging. Assuming that carbon went to 1.3-m tall trees was a simplified solution to ensure carbon conservation without representing the initial growth. 

The allometry problem could be addressed by a seedling-specific set of allometric equations if they exist, but this would require thinking how to make sure the transition from seedling to adult allometry smooth.

**Comment by JunyanDing** [comment]
@mpaiao thanks for the suggestion. Yes, I am aware of the allometric issue of the very small tree. It is a big issue for FATES, even bigger for Hydro! I have to put the initial size of the seedling to be 2m, otherwise the allometry equation works for seedling will cause problems to big ones. Given this, I think the current routine of recruitment is problematic. Also, unless we can set the seedling to be very small, e.g. similar to the seed, the current recruitment routine is not biological sound. 

Or we can use a logistic equation instead of power law function for the allometric relation. That might work for both seedling and big tree.

**Comment by rosiealice** [comment]
Hi @JunyanDing .  Thanks for looking into this. You are right about the nonesensicalness of it all!  

The reason for the current scheme is largely to pass carbon balance checks. If we have the expectation that FATES does not lose or gain carbon then the flux of carbon going into the seed pool has t be the same as that going out, which means that the seedlings are effectively creating by notionally glueing lots of sees together! 

This is of course ridiculous. As far as I can see there are two alternatives:

1. Simulate the actual germination of seeds from their smallest possible size and all of the photosynthesis/respiration that occurs between germination and the attainment of the current 'minimum size' category. Effectively reducing the minimum size so much that each seedline consist of 1 or close-to-1 seed.

2. Break the assumption that we can't get any co2 from the atmosphere and instead use that to make the seedlings up to their minimum size from seeds (which is what I think you are suggesting?

I am not sure what is the right approach.

1) may well add a large computational burden.  Also in reality seedling mortality is very high between germination and the current minimum recruit size, and so we'd either have to ignore that, or somehow model it (in an even-larger data vacuum than we have for trees). Then there are the very-small-plant allometry issues raised already. 

2) will result in a non-trivial carbon flux into the land surface that may or may not be consistent with the  environmental drivers. Although, this will quickly go away if no more seeds are made.  We would need some changes to the carbon balance checks. 
It would be more realistic, as the actual number of germinated seeds would have a meaning (if we are ever in the position of knowing that!).

Another more general problem in FATES has been that the current scheme does not restrict the number of recruits with respect to space (many other models do this), which comes from the fact that we don't want to pre-determine the maximum number of canopy layers (in line with the original intention of the PPA).  What tends to happen now is that if there are too many recruits they fill up the second and third canopy layer and then die slowly of carbon starvation, which makes the results look really strange.  I've never known what the right solution is to this, other than careful oarameterization of the recruitment model. 

I know @lmkueppers, @adamhb  and co. are thinking a lot about this, tagging them so they are in the loop...

**Comment by rgknox** [comment]
@JunyanDing you've brought up some great points, and like others, I think you are spot on regarding the strangeness in how we impose mass conservation between seed mass and recruits.  I feel there is no reason we have to use the seed mass pool to completely build the recruit pool.  

I just wrote a bunch of stuff about a spike flux from the CO2 pool, but now see that Rosie brought up the same points.  I  will just note that this is also true for the nutrient pool.  With a C, N and P model, we also use N and P from the seeds to build the recruits.  So these pools would have to be drawn from the soil if we are not forcing it to come from the seed pool.

**Comment by JunyanDing** [comment]
@rosiealice , thanks for explaining the reason for this weird recruitment routine to me. I feel the 2nd approach is more feasible. The 1st approach means we need to model the whole processes from the germination, and the growth of the baby plant to a successfully established seedling. This approach not only result in high computation cost but also require lot of information about the biological processes in this stage:1)we don’t know the allometry of these very small plant, 2) the mortality of the baby trees under same environment conditions are very different from the established ones as they are more susceptible to harsh condition e.g. wind, storm and frost, like you said “seedling mortality is very high between germination and the current minimum recruit size”. I think we don’t want to explicitly model these processes, but we don’t want to ignore the effect either. I feel make changes on current carbon balance check will be easier to do. Or, we can use the soil pool for the difference in the mass between a single seed and a single seedling for all the elements: C,N,P, since the soil carbon will finally go to atmosphere through soil respiration, this is just like taking a short cut. @rgknox , what do you think? 




> “Another more general problem in FATES has been that the current scheme does not restrict the number of recruits with respect to space (many other models do this), … What tends to happen now is that if there are too many recruits they fill up the second and third canopy layer and then die slowly of carbon starvation” 

Indeed, this has caused problem for my sensitivity analysis of the diagnostic runs of FATES-Hydro. Interestingly, the consequences of having too many recruitments are different depends on the hydraulic strategy, e.g. safe vs. risky stoma. Currently, the recruitment rate is the same as the germination rate. I think to constrain the germination rate by the canopy status, e.g. weight recruitment rate by the canopy spreading variable, sites(s)%spread, e.g. recruitment rate = germination rate * sites(s)%spread, and also reduce the germination rate so that it also take into account the survivorship of the seedling. I think this site_spread variable gives some indirect measure of space limitation of the site.  I feel we need to get both right, otherwise the constrain on the recruitment may result in unwanted effect on population dynamic because of this unrealistic mass transfer routine of the recruitment.

**Comment by rgknox** [comment]
As everyone is indicating, I agree that simulating newly germinated plants from seed to sapling, would require more knowledge about allometry.  We have a hard enough time coming up with appropriate allometric parameterizations for the sapling-to-mature range already. 

My sense is that with our current operating designation of a new recruit, (around 1.3 m tall) plants at that starting size are probably (in many cases) not consistent with the various allometry parameterizations that we are using (which are retrieved from censuses of large trees).   A few years ago I was looking at the data by King et al. at BCI (which has allometry relationships for small plants, sapling size I think.. ).  I found that the King et al data did not really fall along the allometry curves for mature plants in the area.

So a conceptual model that uses co2 and soil nutrients to get us from seed to sapling sounds nice. :)

Back when I was looking at the allometry at BCI and the King dataset, I started to play around with developing routines that would use cubic splines to blend the allometry functions of the treelets/saplings with those of the mature plants.  I think I punted on this because I didn't want to introduce more parameters or requirements to know the allometry of saplings, but I kept the function in there in case it would be useful to blend two different allometries one day:

https://github.com/NGEET/fates/blob/master/biogeochem/FatesAllometryMod.F90#L2463

**Comment by pollybuotte** [comment]
In case people aren't aware, @adamhb is actively working on recruitment code.  Adam, or @lmkueppers maybe you want to chime in?

**Comment by adamhb** [comment]
Hi Polly,
Thanks for alerting me to this. Im in transit today, but will give a proper response tomorrow morning when Im back.
Adam

**Comment by adamhb** [comment]
Hi Junyan and all,

Sorry for the delay. I agree with the issues/observations raised above. This is a problem that Lara and I have been discussing, but its not the focus of the work we are doing now. We have been incorporating PFT-specific environmental sensitivity (light and moisture) into the recruitment process. In our new approach we introduce a seedling pool that is tracked in units of carbon (no seedling allometry). Environmentally sensitive functions mediate the flow of carbon from the seed pool to the (new) seedling pool, and the flux out of the seedling pool (i.e. recruitment). The issues raised above are still issues within this new scheme because seeds effectively "glue together" (to borrow Rosie's imagery) to form seedlings and then seedlings "glue together" again to form saplings. Aside from the issues raised above it makes it hard to relate these parameters to observations of germination or seedling to sapling transition rates because the parameters operate on biologically unrealistic processes.

I think Junyan's proposed general approach to fixing this issue sounds reasonable, and is one that Lara and I have also considered. I'm implementing the new regeneration scheme code such that it can be switched on or off depending on the simulation, so I think fixing this issue in the base regeneration approach wouldn't necessarily mess thing up for our work if Junyan wanted to proceed forward with a fix, but this probably warrants further careful consideration and input from Lara and others. Just thinking off the top of my head I think Junyan's "fix" for this problem could also be implemented in our new regeneration scheme. It would just have to happen twice: once, for the seed to seedling transition and once for the seedling to sapling transition.

I'm happy to keeping chatting more about this and can share a manuscript we have in review that describes the new recruitment approach. Just let me know if I should send it along.
best,
Adam

**Comment by JunyanDing** [comment]
Thanks @adamhb . I can make these changes and do a pull request this week.

**Comment by ckoven** [comment]
Hi all, and thanks @JunyanDing for bringing this up.  At the risk of sounding like I'm defending the status quo (which I don't really want to do), I think I have a slightly different take on the degree of nonsensicalness of the current scheme than @rosiealice, which is that I've interpreted the seed carbon to recruit carbon survival fraction (which is really a way more complicated parameter derived from `EDPftvarcon_inst%germination_rate(pft) / (EDPftvarcon_inst%germination_rate(pft) + EDPftvarcon_inst%seed_decay_rate(pft))`) as being the product of three hidden parameters: the fraction of reproductive allocation that is actual seed mass, the mass ratio between a recruit and a seed, and the survival probability of a seed to a recruit.  As long as the product of those three is less than one, then we can in principle get away with the current scheme—which is really even worse than @rosiealice's description because we are gluing not just a bunch of seeds together but also gluing the cones, flowers, pollen, fruit, etc. to make the recruits—from a carbon budgeting perspective, whereas if it is greater than one, than we do need to make up some carbon somewhere.  I have no idea if that term is in fact always less than one.  I am not sure if it necessarily helps solve the problem to add seed size, seed survival probability, and fraction of reproductive allocation that is seed mass as three distinct parameters, rather than the current scheme that is implicitly (and nowhere actually stated that that is so) based on the product of them all, but maybe it does if it helps clarify how the model is really meant to be structured.  So, all that said, I'm very excited for @adamhb's more realistic scheme, and would be happy to see a more detailed treatment of what our existing scheme is actually doing and what assumptions it is making.  Also, I agree with @rosiealice's point that a very practical problem of the existing scheme is that it fills the understory with recruits just waiting to die, and we should do something about that.

**Comment by adamhb** [comment]
Thanks @ckoven for your thoughts. I just wanted to reiterate that the work I've been doing doesn't attempt to change this aspect of the status quo so a decision on whether to change this or not doesn't impact the work we've already been doing thankfully. I suppose there are tradeoffs between keeping the status quo and implementing something like what @JunyanDing suggested. Maybe its a judgement call on whether or not the benefit of the current simplicity outweighs the cost of making it harder to compare regeneration in FATES to observations? Perhaps, where you fall on that judgement call depends on how much you think getting these regeneration processes correct is important for capturing changes in processes / state variables with more leverage on ecosystem function?
